process/response.go

```diff
@@ -0,0 +1,88 @@
+package process
+
+import (
+	"errors"
+	"github.com/wptide/pkg/payload"
+)
+
+// Response defines the structure for a Response process.
+// This determines where the processed results will be sent.
+type Response struct {
+	Process                                 // Inherits methods from Process.
+	In         <-chan Processor             // Expects a processor channel as input.
+	Out        chan Processor               // (Optional) Send results to an output channel.
+	Payloaders map[string]payload.Payloader // A map of "Payloader"s for different services.
+}
+
+func (res *Response) Run() (<-chan error, error) {
+
+	if res.In == nil {
+		return nil, errors.New("requires a previous process")
+	}
+
+	if len(res.Payloaders) == 0 {
+		return nil, errors.New("need to provide at least one payload manager")
+	}
+
+	errc := make(chan error, 1)
+
+	go func() {
+		defer close(errc)
+		for {
+			select {
+			case in := <-res.In:
+
+				// Copy Process fields from `in` process.
+				res.CopyFields(in)
+
+				// Run the process.
+				// If processing produces an error send it up the error channel.
+				if err := res.process(); err != nil {
+					// Pass the error up the error channel.
+					errc <- err
+					// Don't break, the message is still useful to other processes.
+				}
+
+				// Send process to the out channel.
+				if res.Out != nil {
+					res.Out <- res
+				}
+			case <-res.context.Done():
+				return
+			}
+		}
+
+	}()
+
+	return errc, nil
+}
+
+func (res *Response) process() error {
+
+	payloadType := res.Message.PayloadType
+	if payloadType == "" {
+		// This is temporary, in future there will be no fallback.
+		// Ensure all tasks include the PayloadType.
+		// @todo An empty payloadType should return an error in future.
```

Comment:
Why wouldn't we always default to `tide`?

Reply:
Its too coupled for a package. We can default to `tide` in the services.